java/spring/security/injection/tainted-html-string.yaml

```diff
@@ -32,46 +32,74 @@ rules:
     interfile: true
   mode: taint
   pattern-sources:
-  - patterns:
+  - label: INPUT
+    patterns:
     - pattern-either:
       - pattern-inside: |
-          $METHODNAME(..., @$REQ(...) $TYPE 
-          $SOURCE,...) {
+          $METHODNAME(..., @$REQ(...) $TYPE $SOURCE,...) {
             ...
           }
       - pattern-inside: |
-          $METHODNAME(..., @$REQ $TYPE 
-          $SOURCE,...) {
+          $METHODNAME(..., @$REQ $TYPE $SOURCE,...) {
             ...
           }
     - metavariable-regex:
         metavariable: $REQ
         regex: (RequestBody|PathVariable|RequestParam)
-    - pattern: $SOURCE
+    - focus-metavariable: $SOURCE
+  - label: CONCAT
+    by-side-effect: true
+    requires: INPUT
+    patterns:
+    - pattern-either:
+      - pattern: |
+          "$HTMLSTR" + ...
+      - pattern: |
+          "$HTMLSTR".concat(...)
+      - patterns:
+        - pattern-inside: |
+            StringBuilder $SB = new StringBuilder("$HTMLSTR");
+            ...
+        - pattern: $SB.append(...)
+      - patterns:
+        - pattern-inside: |
+            $VAR = "$HTMLSTR";
+            ...
+        - pattern: $VAR += ...
+      - pattern: String.format("$HTMLSTR", ...)
+      - patterns:
+        - pattern-inside: |
+            String $VAR = "$HTMLSTR";
+            ...
+        - pattern: String.format($VAR, ...)
+    - metavariable-regex:
+        metavariable: $HTMLSTR
+        regex: ^<\w+
+  pattern-propagators:
+  - pattern: (StringBuilder $SB).append($...TAINTED)
```

Comment:
if we're adding propagators why do we need to do the above rules? e.g.$SB.append(...)

Reply:
initially added them to make sure that there's a concatenation with something that looks like a HTML tag that goes into ResponseEntity (lower FPs)